tests/functional/spec/timeout_modal.spec.js

```diff
@@ -0,0 +1,47 @@
+import TimeoutInterstitialPage from "../generated_pages/timeout_modal/timeout-modal-interstitial.page";
+import TimeoutSubmitPage from "../generated_pages/timeout_modal/submit.page";
+import { TimeoutModalPage } from "../base_pages/timeout-modal.page.js";
+
+describe("Timeout Modal", () => {
+  beforeEach("Load the survey", () => {
+    browser.openQuestionnaire("test_timeout_modal.json");
+    browser.setTimeout({ script: 70000 });
+  });
+
+  describe("Given I am completing the survey, when the session timeout is set to 2 minutes", () => {
+    it("I will be able to see the timeout modal with the option to extend the session after 60 seconds", () => {
+      checkTimeoutModal();
+    });
+  });
+
+  describe("Given I am completing the survey, when I click on “Continue survey” button of the timeout modal", () => {
+    it("I will be able to extend the session", () => {
+      checkTimeoutModal();
+      $(TimeoutModalPage.submit()).click();
+      expect($(TimeoutModalPage.timer()).getText()).to.equal("");
```

Comment:
We should assert that:
- the modal is not displayed, then wait like 15 second and ensure it is we are still on the same page (i.e redirect didn't happen)
- to ensure it did actually extend the session, assert that if we refresh, we are still on the same page (i.e it hasn't timed us out in the background).

Reply:
So we ensure now that redirect didn't happen (waiting 15 seconds, had to extend the timeout to 80000). I don't think we can refresh the window using webdriver.